scripts/CMakeLists.txt

```diff
@@ -43,6 +43,18 @@ else()
     message(FATAL_ERROR "Unsupported system.")
 endif()
 
+if(CMAKE_CXX_COMPILER MATCHES ".*icpx" OR CMAKE_CXX_COMPILER MATCHES ".*icx")
+    if(NOT APPLE)

```

Comment:
icx would be normal C++ compiler on linux as well, it's not just sycl compiler. So this logic should be different.  You need fsycl only in case you are going to compile sycl kernels/aplications or link with sycl lib.  but not for cases there icx used as CPP compielr on linux/win